File: internal/doc_engine/grounding/langmatrix/testdata/swift/sample.swift
import Foundation

/// Service settings loaded from the environment.
struct Config { var addr: String; var port: String }

/// Load settings from the environment.
func loadConfig() -> Config {
  let env = ProcessInfo.processInfo.environment
  return Config(addr: env["SAMPLE_ADDR"] ?? "localhost", port: env["SAMPLE_PORT"] ?? "8080")
}

/// Format a greeting; throws on empty input.
func formatGreeting(_ name: String) throws -> String {
  guard !name.isEmpty else { throw NSError(domain: "greet", code: 1) }
  return "hello \(name)"
}

/// Greet one name; keeps a call edge from greetAll.
func greetOne(_ name: String) async throws -> String { try formatGreeting(name) }

/// Greet every name concurrently with structured concurrency.
func greetAll(_ names: [String]) async throws -> [String] {
  try await withThrowingTaskGroup(of: String.self) { group in
    for n in names { group.addTask { try await greetOne(n) } }
    var out: [String] = []
    for try await s in group { out.append(s) }
    return out
  }
}

/// Test-looking function exercising formatGreeting.
func testFormatGreeting() throws { assert(try formatGreeting("ada") == "hello ada") }

File: internal/doc_engine/grounding/langmatrix/testdata/rust/sample.rs
//! Grounding fixture for the depth matrix.
use std::{collections::HashMap, env, sync::{Arc, Mutex}, thread};

/// Service settings loaded from the environment.
pub struct Config { pub addr: String, pub port: String }

/// Load settings from the environment.
pub fn load_config() -> Config {
    Config {
        addr: env::var("SAMPLE_ADDR").unwrap_or_else(|_| "localhost".into()),
        port: env::var("SAMPLE_PORT").unwrap_or_else(|_| "8080".into()),
    }
}

/// Format a greeting; errors on empty input.
pub fn format_greeting(name: &str) -> Result<String, String> {
    if name.is_empty() { return Err("greet: empty name".into()); }
    Ok(format!("hello {name}"))
}

/// Greet one name; keeps a call edge from greet_all.
pub fn greet_one(name: &str) -> Result<String, String> { format_greeting(name) }

/// Greet every name on scoped background threads.
pub fn greet_all(names: Vec<String>) -> Vec<String> {
    let out = Arc::new(Mutex::new(HashMap::new()));
    thread::scope(|s| {
        for n in &names {
            let out = Arc::clone(&out);
            s.spawn(move || { out.lock().unwrap().insert(n.clone(), greet_one(n)); });
        }
    });
    names.iter().map(|n| out.lock().unwrap().remove(n).unwrap().unwrap_or_default()).collect()
}

#[test]
fn test_format_greeting() { assert_eq!(format_greeting("ada").unwrap(), "hello ada"); }
